src/components/API.js

```diff
@@ -25,7 +25,7 @@ export default class Api {
       },
     });
   }
```

Comment:
**Might be better**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` and so on.

Reply:
I'm confused by this comment.  I do not know what you are talking about.

Reply:
> I'm confused by this comment. I do not know what you are talking about.

I wrote you that there was a malfunction that's why I just put it in `Api` but it's for any class. 

In `Card.js` for instance you have this code in line 109

```js

    this._cardElement.querySelector("#card-image").src = this._link;
    this._cardElement.querySelector("#card-name").textContent = this._name;
    this._cardElement.querySelector("#card-image").alt = this._name;

```


You repeat  searching for the image 2 times here and 1 time in line 44, so you need to define a class field


```js

    this._cardImage = this._cardElement.querySelector("#card-image")
    this._cardImage.src = this._link;
    this._cardImage.alt = this._name;
    this._cardElement.querySelector("#card-name").textContent = this._name;

```


and in line 44 ` this._cardImage ` will be accessible